examples/31_custom_drift_models.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+
+# (C) Copyright 2020, 2021, 2022, 2023, 2024 IBM. All Rights Reserved.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""aihwkit example 31: customized conductance drift models
+
+Simple 1-layer network that demonstrates user-defined drift model
+capability and impact on output over time due to drift.
+
+Reference paper evaluating a number of different conductance-dependent
+drift models exhibiting complex characteristics:
+https://onlinelibrary.wiley.com/doi/full/10.1002/aelm.202201190
+"""
+# pylint: disable=invalid-name
+from numpy import asarray
+
+# Imports from PyTorch.
+from torch import (
+    zeros,
+    ones,
+    mean,
+    std,
+    linspace,
+)
+import matplotlib.pyplot as plt
+
+# Imports from aihwkit.
+from aihwkit.nn import AnalogLinear
+from aihwkit.simulator.rpu_base import cuda
+from aihwkit.inference.converter.conductance import SinglePairConductanceConverter
+from aihwkit.inference import PCMLikeNoiseModel
+from aihwkit.simulator.parameters.enums import BoundManagementType
+from aihwkit.simulator.configs import TorchInferenceRPUConfig
+
+g_min, g_max = 0.0, 25.
+# define custom drift model
+custom_drift_model = dict(g_lst=[g_min, 10., g_max],
+                          nu_mean_lst=[0.08, 0.05, 0.03],
+                          nu_std_lst=[0.03, 0.02, 0.01])
+
+t_inference_times = [1,                         # 1 sec
+                     60,                        # 1 min
+                     60 * 60,                   # 1 hour
+                     24 * 60 * 60,              # 1 day
+                     30 * 24 * 60 * 60,         # 1 month
+                     12 * 30 * 24 * 60 * 60,    # 1 year
+                     ]
+
+IN_FEATURES = 512
+OUT_FEATURES = 512
+BATCH_SIZE = 1
+
+# define rpu_config
+rpu_config = TorchInferenceRPUConfig()
+rpu_config.noise_model = PCMLikeNoiseModel(
+    prog_noise_scale=0.0,   # turn off to show drift only
+    read_noise_scale=0.0,   # turn off to show drift only
+    drift_scale=1.0,
+    g_converter=SinglePairConductanceConverter(g_min=g_min, g_max=g_max),
+    custom_drift_model=custom_drift_model)
+rpu_config.forward.bound_management = BoundManagementType.NONE
```

Comment:
It would be better to use the dataclass instantiation way to set the parameter in the example, like: 
```
io_pars = IOParameter(nm_thres=1.0, inp_res=2 **8 -2 ,....)
noise_model = PCMLikeNoiseModel(prog_noise_scale=0.0, ....)
rpu_config = TorchInferenceRPUConfig(noise_model=noise_model, forward=io_pars)
```
This would avoid the case that accidentally wrong assignments are made. Maybe better show-case this "more correct" way of building the `rpu_config`

Reply:
Done!